### ISSUE TITLE
optimize the wasm dockerfile

### ISSUE THREAD

**Issue by gregsadetsky** [gregsadetsky/rapidriteros#16]
heya @bcspragu I noticed in the deployment logs that some stuff gets re-downloaded on deployment by docker

specifically, this line:

https://github.com/gregsadetsky/rapidriteros/blob/main/renderers/wasm/Dockerfile#L33

is not cached in the Dockerfile - I'm not too sure why (i.e. if the wasm source code doesn't change, I think that the docker lines before should be fine? but maybe `adduser` does something to the filesystem that doesn't get cached (that, or the `cargo build --release` which creates files that are somehow... unique?)

in any case, could you try moving the `FROM gcr.io/distroless/cc` much higher in the file?

do you know if you need it actually (i.e. could the initial `FROM rust` be enough)? if you could remove it that could also be interesting and much faster for deployment.

thanks a ton!

**Comment by gregsadetsky** [comment]
according to claude......... (for what it's worth)

you could do

```dockerfile
FROM rust:1.82.0 AS builder
FROM gcr.io/distroless/cc:latest-amd64 AS runtime

# Start with the builder stage
FROM builder
```

and then where you had "FROM gcr..." you could do

```dockerfile
# Switch to the runtime stage
FROM runtime
```

maybe that's worth a try!

**Comment by gregsadetsky** [comment]
fixed by #18!! thank you so much!!!
